src/NuGet.Core/NuGet.Protocol/Resources/PackageSearchResourceV3.cs

```diff
@@ -1,24 +1,43 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
 using System.Linq;
+using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
 using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Model;
 
 namespace NuGet.Protocol
 {
     public class PackageSearchResourceV3 : PackageSearchResource
     {
+        private readonly HttpSource _client;
+        private readonly Uri[] _searchEndpoints;
+
+#pragma warning disable CS0618 // Type or member is obsolete
         private readonly RawSearchResourceV3 _rawSearchResource;
+#pragma warning restore CS0618 // Type or member is obsolete
 
+        [Obsolete("Use PackageSearchResource instead (via SourceRepository.GetResourceAsync<PackageSearchResource>")]
         public PackageSearchResourceV3(RawSearchResourceV3 searchResource)
             : base()
         {
             _rawSearchResource = searchResource;
         }
 
+        internal PackageSearchResourceV3(HttpSource client, IEnumerable<Uri> searchEndpoints)
```

Comment:
This feels like it should be public 

@zivkan thoughts?

Reply:
Actually @zivkan recommended internal.

Reply:
I see. 
Let's leave it then.